common/source/docs/common-hobbywing-dronecan-esc.rst

```diff
@@ -0,0 +1,78 @@
+   :target: ../_images/hobbywing-dronecan-esc.png
+   :width: 450px
+
+Hobbywing ESCs with CAN interfaces (`including these <https://www.hobbywing.com/en/products?id=46>`__) support DroneCAN which allows the autopilot to control the ESC/motor via CAN and also retrieve the individual rpm, voltage, current and temperature.
+
+The contents of this page were verified using a `Hobbywing X6 Plus (6218-150KV CW motor) <https://www.hobbywingdirect.com/collections/xrotor-x8-series/products/xrotor-pro-x6-series>`__.
+
+Some images courtesy of hobbywing.com
+
+Connection and Configuration
+============================
+
+Connect the ESCs (perhaps using a 4-pin I2C splitter) to the autopilot's CAN1 (or CAN2) port.
+
+.. image:: ../../../images/hobbywing-dronecan-esc-autopilot.png
+   :target: ../_images/hobbywing-dronecan-esc-autopilot.png
+   :width: 450px
+
+Connect to the autopilot with a ground station and set the following parameters and then reboot the autopilot
+
+- Set :ref:`CAN_P1_DRIVER<CAN_P1_DRIVER>` to 1 (First Driver)
+- Set :ref:`CAN_D1_PROTOCOL<CAN_D1_PROTOCOL>` to 1 (DroneCAN)
+- Modify :ref:`CAN_D1_UC_OPTION<CAN_D1_UC_OPTION>` so that "Hobbywing ESC" is checked (or directly set to 128)
+
+.. image:: ../../../images/hobbywing-dronecan-esc-options-param.png
+   :target: ../_images/hobbywing-dronecan-esc-options-param.png
+
+Configuring the ESCs
+====================
+
+By default the ESCs come from the factory with an incorrect baudrate (500,000 instead of 1,000,000) and all IDs are set to 1 (the first motor).  The instructions below show how to configure the ESCs to use the correct baudrate and individual ID for each ESC.
+
+You may initially test that the ESCs can be spun in their factory configuration by doing the following:
+
+- Disconnect any other DroneCAN devices from the autopilot's CAN1 port
+- Connect to the autopilot and set :ref:`CAN_P1_BITRATE<CAN_P1_BITRATE>` to 500,000
+- Reboot the autopilot
+- Power on the ESCs, they should stop beeping
+- Use Mission Planner's Setup, Optional Hardware, Motor Test page's "Test All Motors" and all motors should spin
+
+Next correct the configuration of each ESC:
+
+- Ensure the autopilot's :ref:`CAN_P1_BITRATE<CAN_P1_BITRATE>` param is set to 500,000 (reboot the autopilot if modified)
+- Determine the autopilot's SLCAN COM port.  This is normally one higher than the MAVLink COM port.  On CubeOrange set :ref:`SERIAL6_PROTOCOL<SERIAL6_PROTOCOL>` to 2 (MAVLink2) and reboot the autopilot
+- Download and install a the :ref:`DroneCAN Gui<common-uavcan-gui>` (version 1.2.25 or higher)
+- Open the DroneCAN GUI
+
+  - Connect to the SLCAN COM port
+  - Uncheck the checkbox beside the "Set local node ID" field
+  - Select Panels, Hobbywing ESC Panel
+  - From the top table select an ESC
+  - Set the Baudrate to 1,000,000
+  - Set ThrottleID and NodeID to the ESC number.  This ESC number is the :ref:`multicoper motor numbering <copter:connect-escs-and-motors>`.
```

Comment:
it's actually the SERVOn number, which happens to be the ESC number for default layout for a copter

Reply:
ah, very important fix, thanks.